### PR TITLE
lower bound for ReadOptions

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
Test may failed, because ci don't use a specified version of RocksDB.

**Comment by zhangjinpeng1987** [comment]
Wait this [pr](https://github.com/facebook/rocksdb/pull/3199)?

File: tests/test_rocksdb_options.rs [hunk 1]
```diff
@@ -584,6 +584,28 @@ fn test_read_options() {
...
+fn test_readoptions_lower_bound() {
+    let path = TempDir::new("_rust_rocksdb_readoptions_lower_bound").expect("");
+    let db = DB::open_default(path.path().to_str().unwrap()).unwrap();
+
+    db.put(b"k1", b"b").unwrap();
+    db.put(b"k2", b"a").unwrap();
+    db.put(b"k3", b"a").unwrap();
+
+    let mut read_opts = ReadOptions::new();
+    let upper_bound = b"k2";
```
Comment (BusyJay) [inline tests/test_rocksdb_options.rs]: s/upper_bound/lower_bound/

File: tests/test_rocksdb_options.rs [hunk 2]
```diff
@@ -584,6 +584,28 @@ fn test_read_options() {
...
 }
 
+#[test]
+fn test_readoptions_lower_bound() {
+    let path = TempDir::new("_rust_rocksdb_readoptions_lower_bound").expect("");
+    let db = DB::open_default(path.path().to_str().unwrap()).unwrap();
+
+    db.put(b"k1", b"b").unwrap();
+    db.put(b"k2", b"a").unwrap();
+    db.put(b"k3", b"a").unwrap();
```
Comment (siddontang) [inline tests/test_rocksdb_options.rs]: can we test tombstone here?

Comment (zhangjinpeng1987) [reply tests/test_rocksdb_options.rs]: We can't.

**Comment by huachaohuang** [comment]
When will we seek with a key smaller than the lower bound? Is that reasonable?

**Comment by zhangjinpeng1987** [comment]
RocksDB updated with [lower bound](https://github.com/pingcap/rocksdb/pull/6)

**Comment by huachaohuang** [comment]
Can you open another PR to update rocksdb first?

**Pull request closed without merging**